<launch>
	<!-- common -->
    <arg name="on_decription" default="false" />
	<arg name="fixed_frame_name" default="rslidar" />
	<arg name="voxel_leafsize" default="0.10"/>

    <!-- preprocess -->
	<arg name="x_max" default="3.0"/>
	<arg name="x_min" default="-3.0"/>
	<arg name="y_max" default="3.0"/>
	<arg name="y_min" default="-3.0"/>
	<arg name="z_max" default="3.0"/>
	<arg name="z_min" default="-3.0"/>

    <!-- ray ground -->
    <arg name="input_point_topic" default="/pothole/preprocess_cloud" /> <!-- input_point_topic, ground filtering will be performed over the pointcloud in this topic. -->
    <arg name="sensor_height" default="1.6" /><!--  Height of the sensor from the ground -->
    <arg name="clipping_height" default="-1.3" /><!-- Remove Points above this height value (default 0.2 meters) -->
    <arg name="min_point_distance" default="1.0" /><!-- Removes Points closer than this distance from the sensor origin (default 1.85 meters) -->
    <arg name="radial_divider_angle" default="0.08" /><!-- Angle of each Radial division on the XY Plane (default 0.08 degrees)-->
    <arg name="concentric_divider_distance" default="0.01" /><!-- Distance of each concentric division on the XY Plane (default 0.01 meters) -->
    <arg name="local_max_slope" default="15" /><!-- Max Slope of the ground between Points (default 8 degrees) -->
    <arg name="general_max_slope" default="5" /><!-- Max Slope of the ground in the entire PointCloud, used when reclassification occurs (default 5 degrees)-->
    <arg name="min_height_threshold" default="0.05" /><!-- Minimum height threshold between points (default 0.05 meters)-->
    <arg name="reclass_distance_threshold" default="0.2" /><!-- Distance between points at which re classification will occur (default 0.2 meters)-->
    <arg name="no_ground_point_topic" default="/points_no_ground" />
    <arg name="ground_point_topic" default="/points_ground" />
    <!-- ndt -->
	
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find pothole)/rviz/pothole.rviz" />
    
	<node pkg="ndt" type="ndt" name="ndt" args="input:=/points_no_ground" output="screen">
		<param name="on_decription" value="$(arg on_decription)"/>
		<param name="fixed_frame_name" value="$(arg fixed_frame_name)"/>
		<param name="voxel_leafsize" value="$(arg voxel_leafsize)"/>
	</node>

	<node pkg="preprocess" type="preprocess" name="preprocess" args="input:=/rslidar_points" output="screen">
		<param name="on_decription" value="$(arg on_decription)"/>
		<param name="fixed_frame_name" value="$(arg fixed_frame_name)"/>
		<param name="voxel_leafsize" value="$(arg voxel_leafsize)"/>
		<param name="x_max" value="$(arg x_max)"/>
		<param name="x_min" value="$(arg x_min)"/>
		<param name="y_max" value="$(arg y_max)"/>
		<param name="y_min" value="$(arg y_min)"/>
		<param name="z_max" value="$(arg z_max)"/>
		<param name="z_min" value="$(arg z_min)"/>
	</node>

    <node pkg="ray_ground_filter" type="ray_ground_filter" name="ray_ground_filter" output="screen">
        <param name="input_point_topic" value="$(arg input_point_topic)" />
        <param name="sensor_height" value="$(arg sensor_height)" />
        <param name="clipping_height" value="$(arg clipping_height)" />
        <param name="min_point_distance" value="$(arg min_point_distance)" />
        <param name="radial_divider_angle" value="$(arg radial_divider_angle)" />
        <param name="concentric_divider_distance" value="$(arg concentric_divider_distance)" />
        <param name="local_max_slope" value="$(arg local_max_slope)" />
        <param name="general_max_slope" value="$(arg general_max_slope)" />
        <param name="min_height_threshold" value="$(arg min_height_threshold)" />
        <param name="reclass_distance_threshold" value="$(arg reclass_distance_threshold)" />
        <param name="no_ground_point_topic" value="$(arg no_ground_point_topic)" />
        <param name="ground_point_topic" value="$(arg ground_point_topic)" />
    </node>
</launch>
